app/src/main/java/edu/sharif/prj01/rock_paper_scissor/Player.java

```diff
@@ -0,0 +1,51 @@
+import java.util.Random;
+
+public class Player extends Thread {
+    private Weapon weapon;
+
+    public Player() {
+        move();
+    }
+
+    public Weapon getWeapon() {
+        return weapon;
+    }
+
+    @Override
+    public void run() {
+        while (true) {
+            try {
+                Thread.sleep(new Random().nextInt(1000) + 1000);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            move();
+            try {
+                synchronized (this) {
+                    this.wait();
```

Comment:
the player run scenario:
1- wait
2- move

Reply:
Yes. Actually when they move, they wait for their opponent to move as well.